File: src/main/java/de/tectoast/emolga/buttons/buttonsaves/MonData.java
package de.tectoast.emolga.buttons.buttonsaves;

import de.tectoast.jsolf.JSONObject;

import java.util.LinkedHashMap;
import java.util.List;

import static de.tectoast.emolga.commands.Command.toSDName;

public class MonData {

    final LinkedHashMap<String, JSONObject> data = new LinkedHashMap<>();
    final List<JSONObject> list;

    public MonData(List<JSONObject> l) {
        list = l;
        l.forEach(j -> data.put(toSDName(j.getString("name")), j));
    }

    public JSONObject getData(String id) {
        return data.get(id);
    }

    public List<JSONObject> getList() {
        return list;
    }
}

File: src/main/kotlin/de/tectoast/emolga/buttons/buttonsaves/MonData.kt
package de.tectoast.emolga.buttons.buttonsaves

import de.tectoast.emolga.commands.Command
import de.tectoast.jsolf.JSONObject

class MonData(val list: List<JSONObject>) {
    val data: MutableMap<String, JSONObject> = mutableMapOf()

    init {
        list.forEach { data[Command.toSDName(it.getString("name"))] = it }
    }

    fun getData(id: String): JSONObject {
        return data.getValue(id)
    }
}